### PR TITLE
Use ignore_missing_simple when reading ZTF FITS thumbnails with astropy>4.3 and address TNS issues

### PR DESCRIPTION
**Pull request by dmitryduev**
In this PR:

- A recent `astropy` version from pypi (`4.3.post1`) considers ZTF alert thumbnails (stored as compressed FITS) invalid since a required keyword is missing. Added `ignore_missing_simple=True` when reading those in.
- Fixed TNS ingestion issues due to recent developments on the TNS side. The TNS now requires all requests to come with the bot name and id, and an API key, all structured in a peculiar way.

### CODE REVIEW

**Review by acrellin — APPROVED** [review]

**Comment by acrellin** [comment]
Thanks dima, please merge when green!

**Comment by stefanv** [comment]
Did you file an issue about this with astropy?

**Comment by dmitryduev** [comment]
@acrellin I added quite a bit of TNS-related stuff here, so probably worth another look. But the relevant test now passes OK!

**Comment by dmitryduev** [comment]
@stefanv that needs investigation first on our side - quite possible that the ZTF thumbnails have been missing that keyword all the time, but only now astropy starting enforcing a check.

**Comment by dmitryduev** [comment]
@stefanv, yup, there you go: https://github.com/astropy/astropy/pull/10895

**Comment by dmitryduev** [comment]
@acrellin @stefanv I updated the code that reads in the ZTF thumbnails taking astropy/astropy#10895 into account - should now work with `astropy>4.3`, let's see.

**Comment by stefanv** [comment]
I think the version requirement must be updated to say something like >= 4.1.post1?

**Comment by dmitryduev** [comment]
haha, everything worked. @stefanv I did >4.2 (that's what we've been using), should be fine.
@stefanv @acrellin please feel free to merge if now looks good to you.

**Comment by stefanv** [comment]
Trying to understand what's happening here: is Kowalski against a live instance of TNS, set up for testing purposes?

**Comment by dmitryduev** [comment]
Sorry @stefanv, didn't understand the question - could you elaborate please?
The TNS ingestion test tries to grab a tiny bit of most recent data posted to the TNS. The relevant endpoint used to be public, however they have implemented multiple changes in the recent months to the way the data can be accessed. I am just trying to keep up to ensure K always has the most recent TNS data available.

**Comment by stefanv** [comment]
Ah, OK, so instead of making this part of standard Kowalski, why not have the CI emit a config file with the correct entries and then run Kowalski as normal?

**Comment by dmitryduev** [comment]
That would require shell magic beyond my skill, i.e. placing the secrets in the correct spot in the config file... The config is really long and regularly updated, I wouldn't want to just >> it as a text into config.yaml in the GA workflow def.
You're black belt in that stuff, mind assisting? ;)

**Comment by stefanv** [comment]
How about something like this:

`merge_yaml.py`:

```python
#!/bin/env python

import sys
import yaml

if not len(sys.argv) == 3:
    print("Usage: merge_yaml.py file1.yaml file2.yaml")
    print("")
    print("Note that values in file2 will override those in file1.")
    sys.exit(1)

f0 = yaml.load(open(sys.argv[1], 'r'), Loader=yaml.SafeLoader)
f1 = yaml.load(open(sys.argv[2], 'r'), Loader=yaml.SafeLoader)

print(yaml.dump({**f0, **f1}))
```

**Comment by stefanv** [comment]
Alternative would be a tool that works something like this:

```
./yaml_plus.py config.yaml --api-key=kjasdkjasd --password=blah --section.variable=blink
```

**Comment by stefanv** [comment]
Also, afraid I stopped at orange belt ;)

File: kowalski/requirements_api.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 aiofiles>=0.6.0
 aiohttp>=3.7.3
 aiohttp-swagger3==0.5.1
-astropy==4.2
+astropy>=4.2
```
Comment (acrellin) [inline kowalski/requirements_api.txt:4]: 4.2_xx versions are still okay, though?

**Review by acrellin — APPROVED** [review]
Looks good, thanks @dmitryduev !

**Comment by dmitryduev** [comment]
@stefanv I like your second idea a lot, will implement. Thanks!

File: tools/patch_yaml.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    :param input_yaml:
+    :param kwargs:
+    :return:
+    """
+    with open(input_yaml) as yaml_file:
+        content = yaml.load(yaml_file, Loader=yaml.FullLoader)
+
+    if len(kwargs) == 0:
+        return
```
Comment (stefanv) [inline tools/patch_yaml.py:20]: This should probably print the original yaml?

File: kowalski/requirements_api.txt (hunk 1 above)
Comment (dmitryduev) [reply kowalski/requirements_api.txt:4]: Yup.

Comment (dmitryduev) [reply kowalski/requirements_api.txt:4]: Hmm. Is 4.2.1 considered >= 4.2?

**Review by stefanv — APPROVED** [review]

File: kowalski/requirements_api.txt (hunk 1 above)
Comment (acrellin) [reply kowalski/requirements_api.txt:4]: Is that what you want? I thought I saw you specify >=4.3 somewhere else

File: tools/patch_yaml.py (hunk 2 above)
Comment (dmitryduev) [reply tools/patch_yaml.py:20]: I decided to update input in place.

**Pull request merged**